### ISSUE TITLE
Hints do not disappear after scrolling window

### ISSUE THREAD

**Issue by joefiesta** [lydell/LinkHints#29]
Sorry if this should not be considered a bug.   My first day using LinkHints and it is awesome.  

Problem:   when I show hints, they don't disappear if I scroll the window.   While they may still work (I didn't test this), they are no longer where they belong in relationship to the link itself. While this may be by design, if so I think it should be mentioned in YOUR EXTREMELY GOOD documentation!

My guess is that this is not by design but rather would be just another frill to make maintaining the addon more complex.  And that would go against the author's philsophy of keeping this addon SIMPLE!   Yay!

**Comment by lydell** [comment]
Hi!

This is how it currently works:

Every second or something I run through the elements of the hints and re-calculate their positions. If they have moved, I also move the hint. If an element has gone off-screen, I remove the hint. If _all_ elements go off-screen I automatically look for new elements to give hints to.

The intention is to support elements that move for some reason. For example, if a banner loads above some links and push them down. Or if a dropdown of links animates into view. Stuff like that.

As a side effect, it also works for scrolling. Personally, I never scroll once I’ve entered Hints mode, but if you the elements of the hints no longer has the same position on-screen and as such are moved on the next 1-second poll cycle mentioned above.

So … after you scroll the hints _should_ move too (though with a little bit of delay). I’ve noticed them sometimes being a couple of pixels off, but they should be placed roughly like before in relation to their elements.

If this is not how it works for you, you might have found a bug. If so I need more information on how to reproduce.

**Comment by joefiesta** [comment]
Hello Simon,
 
thanks for the very thorough reply, but I think you have a bug.
 
 
My setup:
    1.  Windows 7 professional
    2.  32-bit processor with 4 cores (circa 2012).  Very robust still.
    3.  Firefox 86.0.1.
    4.  Firefox addons:
           profile one has many (about 12) and the other I keep vanilla so
it has only Link Hints at this time
 
     At the time of testing:
            1.  Firefox profile 1 has 3 windows and about 20 tabs.
            2.  Firefox profile 2 has 1 window and 2 tabs
            3.  Firefox is running (ridiculously) some 14 processes.  
 
I started out today checking the issue on the 12-addon profile.  Scrolling a
small amount causes the hints to move, but only after about 4-5 seconds.  I
also found that scrolling a couple pages sometimes fails to update the
hints.
 
I then installed Link Hints on my vanilla profile.  Test showed VERY speedy
hint creation there, but, again when I scroll significant amounts (a page or
two) hints failed to repopulate.  However, what was odd was that AT FIRST
when I first created the hints, it about 4-5 seconds to create them.
 
Now what I am noticing is HIGH CPU utilization.  One of my FF processes is
running at 25% (full capacity for one core for me).    Unfortunately, with
only 4 windows, I see that this process is NOT one of the processes
associated with any of my Firefox windows.  
 
I started closing tabs.
 
I had one window with 4 tabs all associated with Pharmacy sites and covid
vaccination scheduling.    Voila, one of those tabs was the culprit.  As I
have been typing, I was hitting over 75% cpu utlilization at times.   Note
also:  none of the is associated with STORAGE utilization, as I has been
running around 10% all morning. 
 
After closing the 4 tabs in the bad window (and thus, of course, the window
itself), the process that was using all the cpu ended.  
 
So, I was detoured by a BAD WEBSITE.  Bad boy!!!
 
test again....  now with 3 windows and 2 profiles running....
 
Oops.... I go back to my primary window and see I had a CRASHED TAB.  
 
Restored all crashed tabs. 
 
I go back to profile 2.  The repopulation now works quickly, but again for
full page scrolls or more it tends to fail, populating only some of the
newly visible links.  
 
The same is true when I go to profile 1 tabs.  Multiple page scrolls fail to
repopulate, sometime only 1 or 2 links of 30.
 
I have recently had a problem with EBAY.  However, I usually run 4 EBAY tabs
and only one of them is an issue.  I had problems with it with SakaKey a
week ago--which is now GONE!.  And, today, it will not function for LINK
HINTS sometimes.   I have to refresh the tab to get LINK HINTS to work.
You might use that site in your testing:
https://www.ebay.com/sch/Fiesta-Vintage/87/i.html?_from=R40
<https://www.ebay.com/sch/Fiesta-Vintage/87/i.html?_from=R40&_nkw=fiesta+har
lequin+fiestaware&_in_kw=2&_ex_kw=&_sacat=87&_udlo=&_udhi=&_ftrt=901&_ftrv=1
&_sabdlo=&_sabdhi=&_samilow=&_samihi=&_sadis=15&_stpos=19018&_sargn=-1%26sas
lc%3D1&_salic=1&_sop=10&_dmd=1&_ipg=200&_fosrp=1>
&_nkw=fiesta+harlequin+fiestaware&_in_kw=2&_ex_kw=&_sacat=87&_udlo=&_udhi=&_
ftrt=901&_ftrv=1&_sabdlo=&_sabdhi=&_samilow=&_samihi=&_sadis=15&_stpos=19018
&_sargn=-1%26saslc%3D1&_salic=1&_sop=10&_dmd=1&_ipg=200&_fosrp=1
 
This is not something I have reported to Ebay.  Particularly, since I'm not
sure where the fault lies.
 
If you can't recreate my setup or recreate the problem, let me know if there
is any way I can assist you analyzing.  I am, not, however, proficient in
any way with HTML code and only a rank novice with CSS.   I was a mainframe
Cobol and Assembler programmer in my day.  I am, however, pretty good with
much of Windows.   And, I love Autohotkey and REXX.  
 
Once again, my compliments on trying to keep LINK HINTS very, very basic!!!

 
A little more.   I just closed down all my Firefoxes.  And restarted--with
that nasty window I had.    I still have the problem.  I should note,
however, that it does not happen ALL THE TIME.  Some scrolls repopulate.
Most do not.  It also seems  to MAYBE be in some way associated with the
NUMBER of links in view.  When there are more, it fails more often.   Note
that at the top of that Ebay link I gave you, the first few pages will have
dozens if not a hundred links.  
 
Cheers!
 
Joe Petree  
 
 
 
 
 
 
 
 

  _____

**Comment by lydell** [comment]
I tried this page: https://www.ebay.com/sch/Fiesta-Vintage/87/i.html?_from=R40. The lints moved along as expected when I scrolled.

For the rest of what you wrote … I’m sorry, I started skimming a bit because of the length. All I can say for now is that I won’t be doing anything unless you or someone else can come up with clear 1,2,3 instructions for me to follow to reproduce. I know that can be very tricky when there are lots of things involved and it only happens sometimes, but given enough time such things can usually be figured out in the end.

One more thing – I’m curious, why do you scroll after showing the hints?

**Comment by joefiesta** [comment]
Why do I scroll after showing hints?
    
    Because I can, I guess!  
 
Sorry I was so long-winded.   I tend to try to be overly precise. 
 
I still get the problem on the page you used.   Here's the main point:
 
     The slow repopulate  happens all the time.   It takes up to 20 seconds.   And, sometimes while it is doing so, before it displays the new hints, it displays your little smily guy in the center of the page:  ¯\_(ツ)_/¯
 
Do you have Win 7?
 
And, yes, I can live with this, but something is certainly not right. 
 
Joe Petree

  _____

**Comment by lydell** [comment]
> Do you have Win 7?

I hanged on to Windows 7 for as long as I could, but when it became end of life I updated to Windows 10 (it was for free even though it was so long after Microsoft stopped offering me the update).

It definitely looks like something is wrong, yes, so I’ll keep the issue open. But since it doesn’t affect me I’m not going to spend time on it personally (well, unless lots of people report it or something).

Thanks for reporting!

**Comment by joefiesta** [comment]
Thanks.  Being 69 and retired, I am trying to keep Win7 til I die.   I tried
the free Win10.  Hate it.  Hate it.   How do people work without window
borders and titles?

  _____

**Comment by lydell** [comment]
It’s been over a year and I have never seen this happen myself and nobody else has chimed in here, so I’m going to close this one now since there seems to be nothing to do.

**Comment by joefiesta** [comment]
Simon,
I have recently--maybe 6 months now I guess--switched to Windows 7, 64 bit.  
And, of course, changes have been made to Firefox since I opened issue 
no. 29.
I tried today to see if the problem still exists.  (Yes, I rarely scroll with hintsshowing.)  And IT DOES NOT.      Hints do repopulate themselves after I scroll.I can mess up linkhints by trying to do things too fast.  But, I think that may bethat I've switched to an older and slower laptop from a desktop.
And, even though it's still Win7, I see many differences betwee the the 
32-bit and 64-bit.  

So, I find it reasonable you closed this.  I also think Win7 is being used lessand less.  Sad.  I hate 10 and will never get 11. (ha ha.  I'm 70.   I think thatcan happen.)

Joe Petree
 
 
 
-----Original Message-----
